lib/ts/recipe/multifactorauth/recipe.ts

```diff
@@ -0,0 +1,512 @@
+import mfaInfoAPI from "./api/mfaInfo";
+import SessionRecipe from "../session/recipe";
+import { PostSuperTokensInitCallbacks } from "../../postSuperTokensInitCallbacks";
+import { User } from "../../user";
+import { SessionContainerInterface } from "../session/types";
+import RecipeUserId from "../../recipeUserId";
+import Multitenancy from "../multitenancy";
+import Session from "../session";
+import AccountLinkingRecipe from "../accountlinking/recipe";
+import { getUser, listUsersByAccountInfo } from "../..";
+import { Querier } from "../../querier";
+import SessionError from "../session/error";
+import { TenantConfig } from "../multitenancy/types";
+
+export default class Recipe extends RecipeModule {
+    private static instance: Recipe | undefined = undefined;
+    static RECIPE_ID = "multifactorauth";
+
+    getFactorsSetupForUserFromOtherRecipesFuncs: GetFactorsSetupForUserFromOtherRecipesFunc[] = [];
+    getAllFactorsFromOtherRecipesFunc: GetAllFactorsFromOtherRecipesFunc[] = [];
+
+    config: TypeNormalisedInput;
+
+    recipeInterfaceImpl: RecipeInterface;
+
+    apiImpl: APIInterface;
+
+    isInServerlessEnv: boolean;
+
+    querier: Querier;
+
+    constructor(recipeId: string, appInfo: NormalisedAppinfo, isInServerlessEnv: boolean, config?: TypeInput) {
+        super(recipeId, appInfo);
+        this.config = validateAndNormaliseUserInput(config);
+        this.isInServerlessEnv = isInServerlessEnv;
+
+        {
+            let builder = new OverrideableBuilder(RecipeImplementation(this));
+            this.recipeInterfaceImpl = builder.override(this.config.override.functions).build();
+        }
+
+        {
+            let builder = new OverrideableBuilder(APIImplementation());
+            this.apiImpl = builder.override(this.config.override.apis).build();
+        }
+
+        this.querier = Querier.getNewInstanceOrThrowError(recipeId);
+    }
+
+    static getInstanceOrThrowError(): Recipe {
+        if (Recipe.instance !== undefined) {
+            return Recipe.instance;
+        }
+        throw new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?");
+    }
+
+    static getInstance(): Recipe | undefined {
+        return Recipe.instance;
+    }
+
+    static init(config?: TypeInput): RecipeListFunction {
+        return (appInfo, isInServerlessEnv) => {
+            if (Recipe.instance === undefined) {
+                Recipe.instance = new Recipe(Recipe.RECIPE_ID, appInfo, isInServerlessEnv, config);
+
+                // We do not want to add the MFA claim as a global claim (which would make createNewSession set it up)
+                // because we want to add it in the sign-in APIs manually.
+
+                PostSuperTokensInitCallbacks.addPostInitCallback(() => {
+                    SessionRecipe.getInstanceOrThrowError().addClaimValidatorFromOtherRecipe(
+                        MultiFactorAuthClaim.validators.hasCompletedDefaultFactors()
```

Comment:
we should change the name of this validator to be something like `hasCompletedMFARequirementForAuth` cause the recipe function that corresponds to this is `getMFARequirementsForAuth`

Reply:
Done